File: solutions/cpp/darts/1/darts.cpp
#include "darts.h"
#include <cmath>
using namespace std;

namespace darts {

    int score(double x, double y){
        double dist = sqrt(x * x + y * y);

        if(dist > 10.0){
            return 0;
        }else if(dist > 5.0){
            return 1;
        }else if(dist > 1.0){
            return 5;
        }else{
            return 10;
        }
    }
    
}  // namespace darts

File: solutions/kotlin/darts/1/src/main/kotlin/Darts.kt
object Darts {

    fun score(x: Double, y: Double): Int {
        val landingPoint = x * x + y * y

        return when {
            100 >= landingPoint && landingPoint > 25 -> 1
            25 >= landingPoint && landingPoint > 1  -> 5
            1 >= landingPoint -> 10
            else -> 0
        }
    }

    fun score(x: Int, y: Int): Int = score(x.toDouble(), y.toDouble())
    fun score(x: Int, y: Double): Int = score(x.toDouble(), y)
    fun score(x: Double, y: Int): Int = score(x, y.toDouble())
}
